examples/llm/src/models/mosaic_gpt.py

```diff
@@ -66,6 +66,17 @@ def __init__(self, cfg: DictConfig):
         self.embedding_fraction = cfg.get('embedding_fraction', 1)
         assert 0 < self.embedding_fraction <= 1, 'model.embedding_fraction must be between 0 (exclusive) and 1 (inclusive)!'
 
+        # default to PreLN blocks for consistency for now
+        # if PostLN ends up being better, switch this default
+        # user can specify using post_ln
+        gpt_block_type = gpt_blocks.GPTBlock if cfg.get(
+            'norm_location',
+            'pre_ln') == 'pre_ln' else gpt_blocks.GPTBlockPostLN
+        # or, if the initialization is DeepNorm, know we have to use PostLN
+        gpt_block_type = gpt_blocks.GPTBlockPostLN if cfg.get(
+            'param_init_fn',
+            'baseline_') == 'deepnorm_' else gpt_blocks.GPTBlock
```

Comment:
```suggestion
        # if the initialization is DeepNorm, know we have to use PostLN
        if cfg.get('param_init_fn', 'baseline_') == 'deepnorm_':
            raise NotImplementedError(
                f'Deepnorm init is only implemented for PostLN block. Set `norm_location: post_ln ` in model configuration.'
            )
```

Silent model change is danger...